tests/BinaryFormatterTests/WhenWorkingWith_Classes.cs

```diff
@@ -26,6 +29,94 @@ public WithCtor(int i, double d, string s)
             public string String { get; set; }
         }
 
+        abstract class WithVirtualProperties
+        {
+            protected static List<WithVirtualProperties_Element> _elements = new List<WithVirtualProperties_Element>();
+            public static IReadOnlyCollection<WithVirtualProperties_Element> AllElements
+            {
+                get
+                {
+                    return _elements as IReadOnlyCollection<WithVirtualProperties_Element>;
+                }
+            }
+            public static IReadOnlyCollection<WithVirtualProperties_Group> AllGroups
+            {
+                get
+                {
+                    return _elements.Select(gr => gr.Group)
+                        .Distinct()
+                        .ToList() as IReadOnlyCollection<WithVirtualProperties_Group>;
+                }
+            }
+
+            public string Name { get; set; }
+
+            public WithVirtualProperties()
+            {
+            }
+            public WithVirtualProperties(string Name)
+            {
+                this.Name = Name;
+            }
+
+            public override string ToString()
+            {
+                return this.Name;
+            }
+        }
+
+        class WithVirtualProperties_Element : WithVirtualProperties, IDisposable
+        {
+            public WithVirtualProperties_Group Group { get; set; }
+
+            public WithVirtualProperties_Element()
+            {
+
+            }
+            public WithVirtualProperties_Element(string Name, WithVirtualProperties_Group Group):base(Name)
+            {
+                this.Group = Group;
+                _elements.Add(this);
+            }
+
+            public virtual IEnumerable<WithVirtualProperties_Group> Groups {
+                get
+                {
+                    return WithVirtualProperties.AllElements
+                        .Where(el => el == this)
+                        .Select(gr => gr.Group)
+                        .Distinct()
+                        .ToList() as IEnumerable<WithVirtualProperties_Group>;
+                }
+            }
+
+            public void Dispose()
```

Comment:
@lukasz-pyrzyk No, it isn't necessary. I added Dispose for future tests. I will cleanup it and will add by necessary.